packages/server/modules/auth/strategies/azure-ad.js

```diff
@@ -0,0 +1,58 @@
+/* istanbul ignore file */
+'use strict'
+
+const passport = require('passport')
+const OIDCStrategy = require('passport-azure-ad').OIDCStrategy
+const URL = require('url').URL
+const appRoot = require('app-root-path')
+const { findOrCreateUser } = require(`${appRoot}/modules/core/services/users`)
+
+module.exports = async (app, session, sessionStorage, finalizeAuth) => {
+
+  let strategy = new OIDCStrategy({
+    identityMetadata: process.env.AZURE_AD_IDENTITY_METADATA,
+    clientID: process.env.AZURE_AD_CLIENT_ID,
+    responseType: 'code id_token',
+    responseMode: 'form_post',
+    issuer: process.env.AZURE_AD_ISSUER,
+    redirectUrl: new URL('/auth/azure/callback', process.env.CANONICAL_URL).toString(),
+    allowHttpForRedirectUrl: true,
+    clientSecret: process.env.AZURE_AD_CLIENT_SECRET,
+    scope: ['profile', 'email', 'openid'],
+    loggingLevel: "debug",
+    passReqToCallback: true
+  }, async (req, iss, sub, profile, accessToken, refreshToken, done) => {
+    done(null, profile)
+  });
+
+  passport.use(strategy)
+
+  app.get('/auth/azure', session, sessionStorage, passport.authenticate('azuread-openidconnect', { failureRedirect: '/error?message=Failed to authenticate.' }))
+  app.post('/auth/azure/callback',
+    session,
+    passport.authenticate('azuread-openidconnect', { failureRedirect: '/error?message=Failed to authenticate.' }),
+    async (req, res, next) => {
+      const { upn, displayName, _raw } = req.user;
+      let user = {
+        email: upn,
+        name: displayName
+      }
+      if (req.session.suuid) {
+        user.suuid = req.session.suuid
+      }
+      const myUser = await findOrCreateUser({ user: user, rawProfile: _raw })
```

Comment:
This bypasses the invite-only behaviour (which is not documented anywhere properly 😅), but check the google strategy for an example correct flow. 

Currently, if the server is invite-only, the expected behaviour is that unless invited, nobody can create an account. That's why there's a bunch of extra checks that need to go in here, but the google strategy should offer a quick copy paste: https://github.com/specklesystems/speckle-server/blob/63340c78b7f02142fdbc3db2459f5b39dba58a6d/packages/server/modules/auth/strategies/google.js#L30-L59

As I'm looking over the flow in there, i think i've spotted a bug of my very own in there. I'll detail in PR next steps for this.